### PR TITLE
[HOPSWORKS-1353] Flink timer should handle case when service is not installed

### PR DESCRIPTION
**Pull request by robzor92**
…vices are not installed

## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [ ] Adds tests for the submitted changes (for bug fixes & features)
- [x] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java [hunk 1]
```diff
@@ -49,13 +49,19 @@
   private DistributedFsService dfs;
   
   @Schedule(persistent = false,
-    minute = "0",
-    hour = "1")
+    minute = "*",
+    hour = "*", second = "*")
```
Comment (SirOibaf) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java]: Why did you change the frequency?

Comment (robzor92) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java]: Woopsie was using it for testing, should not have been committed

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java [hunk 2]
```diff
@@ -56,6 +56,12 @@ public void deleteOrphanJobs(Timer timer) {
     //Get all jobs from history server
     DistributedFileSystemOps dfso = null;
     try {
+      if(!flinkController.isFlinkInstalled()) {
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java]: @robzor92 @SirOibaf isn't it better to get this information from HostServices?

Comment (robzor92) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java]: That's probably true, sound ok @SirOibaf ?

Comment (SirOibaf) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java]: I think this solution is slightly better. Here you also cover the case in which the history server has been deployed, but Hopsworks for whatever reason doesn't have access to the configuration file

Comment (tkakantousis) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java]: @SirOibaf that reason is an exceptional case, as in an exception would be thrown  because if the service is installed then the directory must be there and if the exception is caught then we can cancel the timer. We wouldn't have defensive code for non-exceptional cases.

**Comment by robzor92** [comment]
https://github.com/logicalclocks/hops-testing/pull/589

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java (hunk 2 above)
Comment (robzor92) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java]: I updated the code, have a look and see what you think. If it is not listed as a service then obviously it should be cancelled. Otherwise any exception happening during exception is logged. I don't think we should add more checks that would cancel the timer, feels like we are over-engineering this.

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java [hunk 3]
```diff
@@ -47,15 +49,25 @@
...
   @EJB
   private DistributedFsService dfs;
-  
+  @EJB
+  private HostServicesFacade hostServicesFacade;
+
   @Schedule(persistent = false,
-    minute = "0",
-    hour = "1")
+    minute = "*",
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/flink/FlinkCleaner.java]: @robzor92  revert
